client/state/data-layer/wpcom-http/pipeline/retry-on-failure/index.jsx

```diff
@@ -3,12 +3,12 @@
 /**
  * External dependencies
  */
-
-import { get, merge } from 'lodash';
+import { get, includes, merge } from 'lodash';
 import { decorrelatedJitter as defaultDelay } from './delays';
 import { default as defaultPolicy } from './policies';
 
-const isGetRequest = request => 'GET' === get( request, 'method', '' ).toUpperCase();
+const isAllowedRequest = ( request, allowedMethods ) =>
+	includes( allowedMethods, get( request, 'method', '' ).toUpperCase() );
```

Comment:
this seems like a reasonable step. I think at the point where we go one or two more exceptions in we may want to pass the request into the policy function and allow them to be more powerful with their decisions.